### PR TITLE
Use a php class for the definitions to avoid loading problems

### PR DESCRIPTION
**Pull request by nickvergessen**
To avoid the "Can not load file ..../definitions.json" we see on drone from time to time.

cc @MorrisJobke @rullzer @LukasReschke

### CODE REVIEW

**Comment by rullzer** [comment]
Ah conflicts!!!!

**Comment by nickvergessen** [comment]
Yeah share by mail was merged, all okay will update tomorrow

**Comment by nickvergessen** [comment]
Rebased

File: lib/private/RichObjectStrings/Validator.php [hunk 1]
```diff
@@ -89,15 +94,16 @@ protected function validateParameter(array $parameter) {
 
 	/**
 	 * @param string $type
+	 * @param array $definition
 	 * @return string[]
 	 */
-	protected function getRequiredParameters($type) {
+	protected function getRequiredParameters($type, $definition) {
```
Comment (LukasReschke) [inline lib/private/RichObjectStrings/Validator.php]: array $definition here? :)

File: lib/public/RichObjectStrings/Definitions.php [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+	 * @return array
+	 * @throws InvalidObjectExeption
+	 * @since 9.2.0
+	 */
+	public function getDefinition($type) {
+		if (isset($this->definitions[$type])) {
+			return $this->definitions[$type];
+		}
+
+		throw new InvalidObjectExeption('Object type is undefined');
```
Comment (LukasReschke) [inline lib/public/RichObjectStrings/Definitions.php]: Missing test. Will add myself :)

**Review by LukasReschke — DISMISSED** [review]
Looks awesome! Two nitpicks, will just adjust them myself and push to this branch :)

🚀 👍

**Comment by LukasReschke** [comment]
LGTM

**Comment by nickvergessen** [comment]
👍 for lukas commit

**Comment by MorrisJobke** [comment]
:+1:

**Pull request merged**